app/src/main/java/org/mozilla/focus/activity/MainActivity.java

```diff
@@ -62,6 +63,11 @@ protected void onCreate(Bundle savedInstanceState) {
 
         final SafeIntent intent = new SafeIntent(getIntent());
 
+        final Set<String> intentCategories = intent.getCategories();
+        if (intentCategories != null && intentCategories.contains(Intent.CATEGORY_LAUNCHER) && intent.getAction().equals(Intent.ACTION_MAIN)) {
```

Comment:
We could move all this inside SafeIntent behind a method. Something like isLauncherIntent() maybe?